src/server/main_service.cc

```diff
@@ -1821,6 +1840,9 @@ void Service::Exec(CmdArgList args, ConnectionContext* cntx) {
   SinkReplyBuilder::ReplyAggregator agg(rb);
   rb->StartArray(exec_info.body.size());
 
+  if (state != ExecEvalState::NONE)
```

Comment:
why not test for a specific state?

Reply:
do we borrow for every multi-exec transaction?

Reply:
`ExecEvalState` show whether it contains any eval, we have ALL, SOME, and NONE. So if we have at least one (i.e. not NONE) we borrow